stablehlo/dialect/TypeInference.cpp

```diff
@@ -866,15 +867,24 @@ LogicalResult inferPadOp(
       return emitOptionalError(
           location,
           "Interior padding cannot be negative: ", paddingInteriorVal);
-    int64_t expectedOutput =
-        inputShape[i] + paddingLowVal + paddingHighVal +
-        std::max<int64_t>(inputShape[i] - 1, 0LL) * paddingInteriorVal;
-    if (expectedOutput < 0)
-      return emitOptionalError(
-          location, "Padding result in negative size for dimension ", i);
-    resultShape.push_back(expectedOutput);
+
+    if (!hlo::isDynamicDimSize(inputShape[i])) {  // static dim
+      int64_t expectedOutput = inferPadDim(inputShape[i], paddingLowVal,
+                                           paddingHighVal, paddingInteriorVal);
+      if (expectedOutput < 0)
+        return emitOptionalError(
```

Comment:
curious if there is some way or plan to share this verifications with op verifiers with some helpers?

Reply:
Not quite follow what you meant here? Please leave comments if the sentences above do not answer your question:
1. For error message level sharing: it is tracked in https://github.com/openxla/stablehlo/issues/154 but we do not have cycles to work on it yet
2. This `expectedOutput < 0` is very specific op logic so I guess no way to do it in ODS?
3. `Op.verify()` are expected to be removed and merged to shape functions (thus shared with MHLO)

Reply:
Oh, the link you posted is of a different op `DynamicPadOp` right? that op will be taken care of in the future. 
This PR is about `PadOp` and I believe there's no `PadOp::verify()` any more (deleted a while ago).